### ISSUE TITLE
Updating loses access to folder and no workflow appears anymore. (99% sure this is a bug)

### ISSUE THREAD

**Issue by stephantual** [11cafe/comfyui-workspace-manager#227]
Last update: February 25th
New update: March 1st.

Upon updating, I noticed ALL my worfklows were gone. Which is a bit of a problem given I rely on workspace manager 'versionning' feature to publish tutorials to youtube and distribute 'step by step' releases.

**I tried to figure out what was going on.** Here's what I can see: 
When i click settings > workspace directory i can see using the 'select folder tool' is pointing to 👍 /C:\AI\ComfyUI_windows_portable\ComfyUI/my_workflows

_This is a weird notation to say the least._ But okay - let's go with that. So i click to my folder anyways by hitting 'reset' and browsing down to my old worfklow folder, press 'save'.

Then i refresh the browser (i even tried restaring the server in case it was a cache thing), and.. still nothing. 

I then thought 'let's save something to see what happens' and sure enough:
OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: '/C:'

which is expected. 

Let's fix it - now let's go in settings again, hit the button that allows us to enter an arbitrary path and enter:
C:\AI\ComfyUI_windows_portable\ComfyUI\my_workflows

Now let's create a new workflow again, hit save - and sure enough it's saved.
BUT - my worflow folder is still empty. This is despite having TONS of versionned flows in there. 

TLDR: while i understand my flows aren't lost, they aren't accessible through the webui. I can open them manually but lost all versions.

Could you please:
a) explain how workspace-manager tracks version
b) give me a hint as to how i could recover my worfklow versions and history?

Thank you!

**Comment by Weixuanf** [comment]
are you in main branch or in beta branch?

**Comment by stephantual** [comment]
Thank you so much for your prompt reply - brilliant extension by the way!
1. main branch - always have been
2. yes, all of them are there. I'm not sure how CWM does the versionning or where it keeps its data, but the wf 'copies' are there.
3. I'm afraid it's entirely blank , and will refill only if i manually load them, save them etc.

**Comment by Weixuanf** [comment]
Did you change browser or cleared your browser cache or anything?

Did your comfyui domain change? For example from 127.0.0.1:8188 to somewhere else, this includes your port change too.

Can you also check comfyui-manager/db/workflowVersions.json is it also empty there?

So all the version data is stored in indexdb, each domain has its own indexdb database, if you manually cleared indexdb (highly unlikely) or changed domain, or changed browser, indexdb data will disappear. In that case, we also have a backup in comfyui-workspace-manager/db/... so you will see .json files there for each table as backup.

But if both of your indexdb and workspace-manager/db is empty....🫠🫠 I'm afraid it's lost.

I'm still trying to figure out why this happened. It seems all you did was updating your workspace manager to latest version?
________________________________

**Comment by stephantual** [comment]
Ahhhhh . okay you nailed it. Right, yeah i changed browser - i thought the db was held on the disk.
Okay - so let's imagine you're using 2 browsers (I need to use FF for recording vids due to the long standing nvidia 'glitch' bug in edge/chrome) - how would i transfer the index db from one to the other or keep them in sync? Any change you could add an option in the settings to 'import' a db?

The reason i ask is that it would be nice to retain versionning.

Cheers!!! 👍

**Comment by Weixuanf** [comment]
>  so let's imagine you're using 2 browsers (I need to use FF for recording vids due to the long standing nvidia 'glitch' bug in edge/chrome) - how would i transfer the index db from one to the other or keep them in sync? 

I can add a "sync db" button in settings, after click, it will sync all your data from the backup data files in `comfyui-workspace-manager/db` to your browser. But you need to make sure `comfyui-workspace-manager/db` wasn't accidentally deleted (some people delete plugin and reinstall sometimes...).

in future we should probably figure out a better way to keep different browser/devices database in sync 🤔 I'm planning to add cloud sync feature so all your data can be synced to cloud server if you choose to login, so all your comfyui clients will have the same data

**Comment by stephantual** [comment]
Thank you! I like physical files and i back up my comfy everynight (i have a bit of a beast of a rig/internet connection) - but yeah makes sense! thank you! closing now as it's resolved!
